ee/hogai/graph/root/nodes.py

```diff
@@ -293,5 +293,5 @@ def router(self, state: AssistantState) -> RouteName:
         if isinstance(last_message, AssistantToolCallMessage):
             return "root"  # Let the root either proceed or finish, since it now can see the tool call result
         if state.root_tool_call_id:
-            return "insights" if state.root_tool_insight_type else "search_documentation"
```

Comment:
Would be clearest if this path also was named afer the tool, i.e. `create_and_query_insight` – the way `search_documentation` already is